internal/pkg/config/output.go

```diff
@@ -228,3 +232,52 @@ func makeURL(defaultScheme string, defaultPath string, rawURL string, defaultPor
 	addr.Host = host + ":" + port
 	return addr.String(), nil
 }
+
+func (c *Elasticsearch) DiagRequests(ctx context.Context) []byte {
+	pURL, err := httpcommon.NewProxyURIFromString(c.ProxyURL)
+	if err != nil {
+		zerolog.Ctx(ctx).Warn().Err(err).Msg("Unable to transform proxy_url to url.URL")
+	}
+	settings := httpcommon.HTTPTransportSettings{
+		TLS:     c.TLS,
+		Timeout: c.Timeout,
+		Proxy: httpcommon.HTTPClientProxySettings{
+			Disable: c.ProxyDisable,
+			URL:     pURL,
+			Headers: httpcommon.ProxyHeaders(c.ProxyHeaders),
+		},
+	}
+	headers := http.Header{}
+	for k, v := range c.Headers {
+		headers.Set(k, v)
+	}
+
+	reqs := make([]*http.Request, 0, len(c.Hosts))
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	var res bytes.Buffer
+	for _, host := range c.Hosts {
+		u, err := url.Parse(host)
+		if err != nil {
+			zerolog.Ctx(ctx).Warn().Err(err).Str("host", host).Msg("Unable to transform host to url.URL")
+			res.WriteString(fmt.Sprintf("Unable to transform host %q to url.URL: %v\n", host, err))
+			continue
+		}
+		if u.Scheme == "" {
+			u.Scheme = c.Protocol
+		}
+		ctx, cancel := context.WithTimeout(ctx, time.Second*30) // TODO(michel-laterman): Configurable timeout; should this be part of config, or part of a diagnostics action?
```

Comment:
As per the discussion in https://github.com/elastic/elastic-agent/issues/4880 i'm adding a timeout to each request so the (diagnostics) action does not indefinitely block.
We can add a `duration` value to the fleet-server action definition in `openapi.yml`
@blakerouse, What do you think about adding a duration attribute to the action definition in elastic-agent-client spec (where we can also use it for the optional CPU trace)?